src/appleseed/renderer/kernel/intersection/intersector.h

```diff
@@ -32,7 +32,9 @@
 
 // appleseed.renderer headers.
 #include "renderer/kernel/intersection/curvetree.h"
+#ifdef APPLESEED_WITH_EMBREE
 #include "renderer/kernel/intersection/embreescene.h"
```

Comment:
Again, I vote for no checks in outer code, check is done inside EmbreeScene.h

Reply:
I prefer them outside myself, it is more readable for me. But it is not a big preference.
What I don't like is that we check outside for some optional dependencies and inside for embree.

Reply:
Okay, then you're safe to remove this check inside the header itself